src/main/java/net/sacredlabyrinth/phaed/simpleclans/hooks/DiscordHook.java

```diff
@@ -0,0 +1,449 @@
+import net.sacredlabyrinth.phaed.simpleclans.managers.ClanManager;
+import net.sacredlabyrinth.phaed.simpleclans.managers.SettingsManager;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.*;
+import java.util.*;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+import static github.scarsz.discordsrv.dependencies.jda.api.Permission.VIEW_CHANNEL;
+import static net.sacredlabyrinth.phaed.simpleclans.ClanPlayer.Channel.CLAN;
+import static net.sacredlabyrinth.phaed.simpleclans.SimpleClans.lang;
+import static net.sacredlabyrinth.phaed.simpleclans.chat.SCMessage.Source.DISCORD;
+import static net.sacredlabyrinth.phaed.simpleclans.hooks.DiscordHook.PermissionAction.ADD;
+import static net.sacredlabyrinth.phaed.simpleclans.hooks.DiscordHook.PermissionAction.REMOVE;
+import static net.sacredlabyrinth.phaed.simpleclans.managers.SettingsManager.ConfigField.*;
+
+/**
+ * Hooks SimpleClans and Discord, using DiscordSRV.
+ * <p>
+ * On server' startup:
+ * </p>
+ * <ul>
+ *  <li>Creates categories and channels, respecting discord's limits.</li>
+ *  <li>Removes invalid channels, resets permissions.</li>
+ * </ul>
+ * <p>
+ * Manages events:
+ * </p>
+ * <ul>
+ *     <li>Clan creation/deletion</li>
+ *     <li>ClanPlayer joining/resigning</li>
+ *     <li>Player linking</li>
+ * </ul>
+ * <p>
+ * Currently, works with clan chat only.
+ */
+public class DiscordHook implements Listener {
+
+    private final SimpleClans plugin;
+    private final SettingsManager settingsManager;
+    private final ChatManager chatManager;
+    private final ClanManager clanManager;
+    private final AccountLinkManager accountManager = DiscordSRV.getPlugin().getAccountLinkManager();
+
+    private final Guild guild = DiscordSRV.getPlugin().getMainGuild();
+    private final List<String> textCategories;
+    private final List<String> discordClanTags;
+    private final List<String> clanTags;
+
+    private final Role leaderRole;
+
+    private static final int MAX_CHANNELS_PER_CATEGORY = 50;
+    private static final int MAX_CHANNELS_PER_GUILD = 500;
+
+    public DiscordHook(SimpleClans plugin) {
+        this.plugin = plugin;
+        settingsManager = plugin.getSettingsManager();
+        chatManager = plugin.getChatManager();
+        clanManager = plugin.getClanManager();
+        textCategories = settingsManager.getStringList(DISCORDCHAT_TEXT_CATEGORY_IDS).stream().
+                filter(this::categoryExists).collect(Collectors.toList());
+
+        List<String> whitelist = settingsManager.getStringList(DISCORDCHAT_TEXT_WHITELIST);
+
+        clanTags = clanManager.getClans().stream().
+                filter(Clan::isVerified).
+                filter(clan -> !getDiscordPlayersId(clan.getTag()).isEmpty() || clan.isPermanent()).
```

Comment:
`!getDiscordPlayersId(clan.getTag()).isEmpty()`
If a Player joins the clan later, there won't be a channel created.

Reply:
Isn't it cotrolled by 
```java
    @EventHandler
    public void onClanCreate(CreateClanEvent event) {
        createChannel(event.getClan().getTag());
    }
    ```
    :eyes:

Reply:
Let's see...
A linked player joined to a not linked clan. Yep, right now, channel is not created. 
But after a server restart it will be created. 👀
 
And, perhaps, it's a good idea to leave it. Imagine, some linked player joined and leaved from 10 clans, it will create 10 channels (if they weren't linked before) 👀